x-pack/plugins/lists/server/services/lists/client.ts

```diff
@@ -64,47 +61,39 @@ import {
   UpdateListOptions,
 } from './client_types';
 
-// TODO: Consider an interface and a factory
 export class ListClient {
-  private readonly spaces: SpacesServiceSetup | undefined | null;
+  private readonly spaceId: string;
+  private readonly user: string;
   private readonly config: ConfigType;
-  private readonly dataClient: Pick<
-    ScopedClusterClient,
-    'callAsCurrentUser' | 'callAsInternalUser'
-  >;
-  private readonly request: KibanaRequest;
-  private readonly security: SecurityPluginSetup;
-
-  constructor({ request, spaces, config, dataClient, security }: ConstructorOptions) {
-    this.request = request;
-    this.spaces = spaces;
+  private readonly callAsCurrentUser: APICaller;
```

Comment:
nit: Naming this `callAsCurrentUser` makes sense for the use case when we are pulling the client scoped to the user off of the request, however when I use the list client in the alerting framework I will be passing down the `callAsInternalUser` function to the list client. This could be confusing to the developer having this function named `callAsCurrentUser` when the api caller might be scoped to the internal user. I think renaming this to something more generic like `callCluster` could alleviate this possible future problem. I am also open to doing the renaming when I open my PR with the changes to the lists setup function and looking at it then.

Reply:
I agree, I will change the name of it and then do a check-in to avoid confusion.